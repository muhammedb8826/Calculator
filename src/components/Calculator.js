import { useState } from 'react';
import calculate from '../logic/calculate';

export default function Calculator() {
  const [obj, setObj] = useState({
    total: '',
    next: '',
    operation: '',
  });

  const calculations = (buttonName) => {
    const res = calculate(obj, buttonName);
    setObj({
      next: res.next,
      total: res.total,
      operation: res.operation,
    });
  };

  return (
    <div className="calculator-grid">
      <div className="output">
        <div className="previous-operand">
          {obj.total}
          {' '}
          {obj.operation}
        </div>
        <div className="current-operand">
          {obj.next}
        </div>
      </div>
      <button
        type="button"
        onClick={() => calculations('AC')}
      >
        AC
      </button>
      <button type="button" onClick={() => calculations('+/-')}>+/-</button>
      <button type="button" onClick={() => calculations('%')}>%</button>
      <button type="button" className="operator" onClick={() => calculations('รท')}>รท</button>
      <button type="button" onClick={() => calculations('7')}>7</button>
      <button type="button" onClick={() => calculations('8')}>8</button>
      <button type="button" onClick={() => calculations('9')}>9</button>
      <button type="button" className="operator" onClick={() => calculations('x')}>x</button>
      <button type="button" onClick={() => calculations('4')}>4</button>
      <button type="button" onClick={() => calculations('5')}>5</button>
      <button type="button" onClick={() => calculations('6')}>6</button>
      <button type="button" className="operator" onClick={() => calculations('-')}>-</button>
      <button type="button" onClick={() => calculations('1')}>1</button>
      <button type="button" onClick={() => calculations('2')}>2</button>
      <button type="button" onClick={() => calculations('3')}>3</button>
      <button type="button" className="operator" onClick={() => calculations('+')}>+</button>
      <button type="button" className="span-two" onClick={() => calculations('0')}>0</button>
      <button type="button" onClick={() => calculations('.')}>.</button>
      <button type="button" className="operator" onClick={() => calculations('=')}>=</button>
    </div>
  );
}
