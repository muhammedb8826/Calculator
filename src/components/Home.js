export default function Home() {
  return (
    <div className="home">
      <h1>Welcome to our page!</h1>
      <p>Lorem</p>
      <p>Lorem</p>
    </div>
  );
}
