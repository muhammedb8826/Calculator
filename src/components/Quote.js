import { useState, useEffect } from 'react';

function Quote() {
  const [data, setData] = useState([]);
  const [hasError, setHasError] = useState(false);
  const [isLoading, setIsLoading] = useState(false);

  useEffect(() => {
    const fetchData = async () => {
      setIsLoading(true);
      try {
        const url = 'https://api.api-ninjas.com/v1/quotes';
        const res = await fetch(url, {
          headers: {
            'Content-Type': 'application/json',
            'x-api-key': 'XZBOGakGnoeKaEBsEJ2C0g==SlYB0NPsstmZBa5G',
          },
        });
        const json = await res.json();
        setData(json[0]);
      } catch (error) {
        setHasError(true);
      }
      setIsLoading(false);
    };
    fetchData();
  }, [setData, setIsLoading]);

  if (hasError) return <div>Something went wrong!</div>;

  if (isLoading) return <div>Loading...</div>;

  return (
    <ul className="fetched-data">
      <li>
        {data.quote}
        {' - '}
        {data.author}
      </li>
    </ul>
  );
}

export default Quote;
