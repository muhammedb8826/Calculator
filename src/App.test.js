import renderer from 'react-test-renderer';
import calculate from './logic/calculate';
import operate from './logic/operate';
import Calculator from './Components/Calculator';
import Home from './Components/Home';
import Quote from './Components/Quote';

describe('Test for calculate', () => {
  it('4 + 4 = 8', () => {
    const obj = {
      total: '4',
      next: '4',
      operation: '+',
    };
    const res = calculate(obj, '=');
    expect(res.total).toBe('8');
  });
  it('4 - 4 = 0', () => {
    const obj = {
      total: '4',
      next: '4',
      operation: '-',
    };
    const res = calculate(obj, '=');
    expect(res.total).toBe('0');
  });
  it('4 x 4 = 16', () => {
    const obj = {
      total: '4',
      next: '4',
      operation: 'x',
    };
    const res = calculate(obj, '=');
    expect(res.total).toBe('16');
  });
  it('4 รท 4 = 1', () => {
    const obj = {
      total: '4',
      next: '4',
      operation: 'รท',
    };
    const res = calculate(obj, '=');
    expect(res.total).toBe('1');
  });
});

describe('Test for operate', () => {
  it('4 + 5 equals 9', () => {
    const result = operate(4, 5, '+');
    expect(result).toBe('9');
  });
  it('4 - 5 equals -1', () => {
    const result = operate(4, 5, '-');
    expect(result).toBe('-1');
  });
  it('4 x 5 equals 20', () => {
    const result = operate(4, 5, 'x');
    expect(result).toBe('20');
  });
  it('4 รท 5 equals 0.8', () => {
    const result = operate(4, 5, 'รท');
    expect(result).toBe('0.8');
  });
  it('4 % 5 equals 4', () => {
    const result = operate(4, 5, '%');
    expect(result).toBe('4');
  });
});

describe('Components render Correctly', () => {
  it('Calculator renders correctly', () => {
    const tree = renderer
      .create(<Calculator />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Home renders correctly', () => {
    const tree = renderer
      .create(<Home />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Quote renders correctly', () => {
    const tree = renderer
      .create(<Quote />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
