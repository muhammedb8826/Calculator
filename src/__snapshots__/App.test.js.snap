// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components render Correctly Calculator renders correctly 1`] = `
<div
  className="calculator-container"
>
  <h1>
    Lets do some math!
  </h1>
  <div
    className="calculator-grid"
  >
    <div
      className="output"
    >
      <div
        className="previous-operand"
      >
         
      </div>
      <div
        className="current-operand"
      />
    </div>
    <button
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="operator"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="operator"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="operator"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="operator"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="span-two"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="operator"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;

exports[`Components render Correctly Home renders correctly 1`] = `
<div
  style={
    Object {
      "padding": "1rem",
    }
  }
>
  <h1>
    Welcome to our page!
  </h1>
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
  </p>
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
  </p>
</div>
`;

exports[`Components render Correctly Quote renders correctly 1`] = `
<ul
  className="fetched-data"
>
  <li>
     - 
  </li>
</ul>
`;
